### PR TITLE
Improve & cleanup documentation comments

### PR DESCRIPTION
**Pull request by cr7pt0gr4ph7**
This PR tries to improve the code documentation by:
* Adding documentation to many fields.
* Using a consistent interpunctuation (Type/Method/.. comments always end with a colon, single line field comments don't).
* Only in a few cases: Reorder struct fields to improve readability.

### CODE REVIEW

File: types/client.go [hunk 1]
```diff
@@ -12,113 +10,88 @@ import (
 
 // CheckpointCreateOptions holds parameters to create a checkpoint from a container
 type CheckpointCreateOptions struct {
-	CheckpointID string
-	Exit         bool
+	CheckpointID string // ID of the created checkpoint
+	Exit         bool   // Whether to exit the container after the checkpoint was was successfully created
```
Comment (allencloud) [inline types/client.go]: There are 2 was, one additional.

**Comment by allencloud** [comment]
I am not a maintainer of this. While I like the purpose very much. API comments help the code be more readable.

File: types/client.go [hunk 2]
```diff
@@ -12,113 +10,88 @@ import (
...
-	Author    string
-	Changes   []string
-	Pause     bool
-	Config    *container.Config
+	Pause     bool              // Pause container during commit
+	Reference string            // Repository (and optionally tag) for the new image
+	Author    string            // Author of the new image
+	Comment   string            // Comment for the new image
+	Changes   []string          // Dockerfile instructions to apply while committing
+	Config    *container.Config // Container configuration for the new image
```
Comment (MHBauer) [inline types/client.go]: Does the order of members matter in golang? Why the change of order?

File: types/client.go [hunk 3]
```diff
@@ -12,113 +10,88 @@ import (
...
 }
 
 // ContainerExecInspect holds information returned by exec inspect.
 type ContainerExecInspect struct {
-	ExecID      string
-	ContainerID string
-	Running     bool
-	ExitCode    int
+	ExecID      string // ID of the exec instance
+	ContainerID string // ID of the container
```
Comment (MHBauer) [inline types/client.go]: these two comments don't really add anything beyond the variable names.

File: types/client.go [hunk 4]
```diff
@@ -12,113 +10,88 @@ import (
...
+	Running     bool   // Whether the exec'ed process is running
+	ExitCode    int    // Exit code of the exec'ed process
```
Comment (MHBauer) [inline types/client.go]: Neither do these two, tbh.

File: types/client.go [hunk 5]
```diff
@@ -12,113 +10,88 @@ import (
...
 type ContainerListOptions struct {
-	Quiet  bool
-	Size   bool
-	All    bool
-	Latest bool
-	Since  string
-	Before string
-	Limit  int
-	Filter filters.Args
+	Quiet  bool         // Only display numeric container IDs
```
Comment (MHBauer) [inline types/client.go]: :+1:

File: types/client.go [hunk 6]
```diff
@@ -12,113 +10,88 @@ import (
...
+	Size   bool         // Show container sizes in the output
+	All    bool         // Also show non-running containers
+	Latest bool         // Show recently created containers (including non-running ones)
+	Since  string       // Only show containers created since ID (including non-running ones)
+	Before string       // Only show containers created before ID (including non-running ones)
+	Limit  int          // Show at most N recently created containers (including non-running ones)
```
Comment (MHBauer) [inline types/client.go]: For all of these, it's not really 'display' or 'show', it's 'return'.

**Comment by allencloud** [comment]
Conflict happens. Please rebase and fix it. @cr7pt0gr4ph7

**Comment by vdemeester** [comment]
ping @cr7pt0gr4ph7

**Comment by crosbymichael** [comment]
Thanks for this PR.

We have moved the code from this repo back into the main docker/docker repository; if you still wish to make this change you will need to reopen this PR on the main repo.  

You can find the code for the client and types here:

https://github.com/docker/docker/tree/master/api/types
https://github.com/docker/docker/tree/master/client

Sorry for the inconvenience but making this change helps people better contribute and not have the vendor mess going forward when changing the types or client for Docker.  

Sorry about that and feel free to ask if you have any questions.
